$OpenBSD: patch-parser_c,v 1.14 2015/11/05 14:24:12 jasper Exp $

Security fix for CVE-2015-7941, upstream git:
a7dfab7411cbf545f359dd3157e5df1eb0e7ce31
9b8512337d14c8ddf662fcb98b0135f225a1c489
41ac9049a27f52e7a1f3b341f8714149fc88d450

Security fix for CVE-2015-7942, upstream git:
bd0526e66a56e75a18da8c15c4750db8f801c52d

From f65128f38289d77ff322d63aef2858cc0a819c34 Mon Sep 17 00:00:00 2001
From: Daniel Veillard <veillard@redhat.com>
Date: Fri, 17 Oct 2014 17:13:41 +0800
Subject: Revert "Missing initialization for the catalog module"

--- parser.c.orig	Thu Nov  5 15:14:01 2015
+++ parser.c	Thu Nov  5 15:15:22 2015
@@ -6905,7 +6905,7 @@ xmlParseConditionalSections(xmlParserCtxtPtr ctxt) {
 				 NULL, NULL);
 	}
 	if ((ctxt-> instate != XML_PARSER_EOF) &&
-	    ((ctxt->input->cur + 3) < ctxt->input->end))
+	    ((ctxt->input->cur + 3) <= ctxt->input->end))
 	SKIP(3);
     }
 }
